src/celeritas/neutron/interactor/ChipsNeutronElasticInteractor.hh

```diff
@@ -0,0 +1,157 @@
+ * G4ChipsElasticModel and G4ChipsNeutronElasticXS, as partly documented
+ * in section 21.1.3 of the Geant4 Physics Reference (release 11.2).
+ */
+class ChipsNeutronElasticInteractor
+{
+    //!@{
+    //! \name Type aliases
+    using Energy = units::MevEnergy;
+    using Mass = units::MevMass;
+    using Momentum = units::MevMomentum;
+    //!@}
+
+  public:
+    // Construct from shared and state data
+    inline CELER_FUNCTION
+    ChipsNeutronElasticInteractor(NeutronElasticRef const& shared,
+                                  ParticleTrackView const& particle,
+                                  Real3 const& inc_direction,
+                                  IsotopeView const& target);
+
+    // Sample an interaction with the given RNG
+    template<class Engine>
+    inline CELER_FUNCTION Interaction operator()(Engine& rng);
+
+  private:
+    //// TYPES ////
+
+    using UniformRealDist = UniformRealDistribution<real_type>;
+
+    //// DATA ////
+
+    // Constant shared data
+    NeutronElasticRef const& shared_;
+    // Incident neutron energy
+    units::MevEnergy const inc_energy_;
+    // Incident neutron direction
+    Real3 const& inc_direction_;
+    // Target nucleus
+    IsotopeView const& target_;
+
+    // Value of neutron MeVMass
+    Mass neutron_mass_;
+    Momentum neutron_p_;
+
+    // Sampler
+    UniformRealDist sample_phi_;
+    detail::MomentumTransferSampler sample_momentum_square_;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with shared and state data, and a target nucleus.
+ */
+CELER_FUNCTION ChipsNeutronElasticInteractor::ChipsNeutronElasticInteractor(
+    NeutronElasticRef const& shared,
+    ParticleTrackView const& particle,
+    Real3 const& inc_direction,
+    IsotopeView const& target)
+    : shared_(shared)
+    , inc_energy_(particle.energy())
+    , inc_direction_(inc_direction)
+    , target_(target)
+    , neutron_mass_(shared_.neutron_mass)
+    , neutron_p_(particle.momentum())
+    , sample_phi_(0, 2 * constants::pi)
+    , sample_momentum_square_(shared_, target_, neutron_p_)
+{
+    CELER_EXPECT(particle.particle_id() == shared_.ids.neutron);
+    CELER_EXPECT(inc_energy_ > zero_quantity());
+}
+//---------------------------------------------------------------------------//
+/*!
+ * Sample the final state of the neutron-nucleus elastic scattering.
+ */
+template<class Engine>
+CELER_FUNCTION Interaction ChipsNeutronElasticInteractor::operator()(Engine& rng)
+{
+    // Scattered neutron with respect to the axis of incident direction
+    Interaction result;
+
+    // The momentum magnitude in the c.m. frame
+    real_type target_mass = value_as<Mass>(target_.nuclear_mass());
+    real_type neutron_mass = value_as<Mass>(neutron_mass_);
+    real_type neutron_energy = neutron_mass + inc_energy_.value();
```

Comment:
Maybe the member variables should be `real_type neutron_mass_` (instead of `Mass`) and `real_type neutron_energy_` (instead of `inc_energy_`, which isn't used anywhere else)?

Reply:
Changed them as suggested.